kernelci/cli/docker.py

```diff
@@ -0,0 +1,203 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+#
+# Copyright (C) 2022, 2023 Collabora Limited
+# Author: Guillaume Tucker <guillaume.tucker@collabora.com>
+# Author: Jeny Sadadia <jeny.sadadia@collabora.com>
+
+"""Tool to generate Dockerfiles from Jinja2 templates and build images"""
+
+
+import io
+import json
+import sys
+
+from jinja2 import Environment, FileSystemLoader
+import docker
+
+from .base import Command, sub_main
+
+
+class DockerCommand(Command):
+    """Base command class for docker sub-commands"""
+
+    args = Command.args + [
+        {
+            'name': 'image',
+            'help': "Docker image name, e.g. gcc-10:x86",
+        },
+    ]
+    opt_args = Command.opt_args + [
+        {
+            'name': '--arch',
+            'help': "CPU architecture, e.g. x86",
+        },
+        {
+            'name': '--fragment',
+            'action': 'append',
+            'help': "Extra fragments, e.g. kernelci",
+        },
+        {
+            'name': '--image-version',
+            'help': "Docker image version tag e.g. 20221011.0",
+        },
+        {
+            'name': '--prefix',
+            'default': 'kernelci/',
+            'help': "Docker image tag prefix",
+        },
+    ]
+
+    @classmethod
+    def _gen_image_name(cls, args):
+        base_name = args.prefix + args.image
+        tag_strings = ([args.arch] if args.arch else []) + (
+            args.fragment or [])
+        if args.image_version:
+            tag_strings.append(args.image_version)
+        tag_name = '-'.join(tag_strings)
+        name = ':'.join((base_name, tag_name)) if tag_name else base_name
+        return base_name, tag_name, name
+
+
+class DockerBuildGenerateCommand(DockerCommand):
+    """Base command class for docker generate and build sub-commands"""
+
+    TEMPLATE_PATH = 'config/docker'
+
+    @classmethod
+    def _get_template_params(cls, prefix, fragment):
+        params = {
+                'prefix': prefix,
+                'fragments': [
+                    f'fragment/{fragment}.jinja2'
+                    for fragment in fragment
+                ] if fragment else []
+            }
+        return params
+
+    def _gen_dockerfile(self, img_name, params):
+        jinja2_env = Environment(loader=FileSystemLoader(self.TEMPLATE_PATH))
+        template = jinja2_env.get_template(f"{img_name}.jinja2")
+        return template.render(params)
+
+    def _get_dockerfile(self, args):
+        params = self._get_template_params(args.prefix, args.fragment)
+        template = '-'.join(
+            (args.image, args.arch)) if args.arch else args.image
```

Comment:
Maybe it's just me but I find it confusing to have an open bracket and the closing one in the middle of the next line.  This would be easier to read I think:
```python
        template = (
            '-'.join((args.image, args.arch)) if args.arch else args.image
        )
```